integration-tests/test_selinux.py

```diff
@@ -0,0 +1,212 @@
+        5. Run insights-client --register as root
+        6. Check that insights-core code was executed in unconfined_t context
+        7. Check that there was no SELinux denial during execution
+    :expectedresults:
+        1. System is registered (handled by register_subman fixture)
+        2. Current SELinux mode is saved
+        3. SELinux is set to enforcing mode
+        4. Current context is confirmed as unconfined_t
+        5. insights-client --register completes successfully
+        6. insights-core runs in unconfined_t context (no transition to confined)
+        7. No SELinux denials are found in audit logs
+    """
+    original_mode = _get_selinux_mode()
+    if original_mode == "Disabled":
+        pytest.skip(
+            reason="Skipping test: SELinux is disabled on this system "
+            "(getenforce returned 'Disabled' or command failed)"
+        )
+
+    try:
+        # Setup: Switch to enforcing mode
+        _set_selinux_mode("Enforcing")
+        assert _get_selinux_mode() == "Enforcing"
+
+        # Verify unconfined_t context (test runs in unconfined_t by default)
+        context = _get_current_context()
+        assert "unconfined_t" in context, f"Expected unconfined_t, got: {context}"
+
+        # Test: Run insights-client in unconfined_t context and check for denials
+        start_time = time.time()
+        # Pass the unconfined_t context to ensure the command runs in that context
+        result = insights_client.run("--register", selinux_context=context)
+
+        def check_registered_with_context():
+            status_result = insights_client.run(
+                "--status", check=False, selinux_context=context
+            )
+            if status_result.returncode == 0 and any(
+                i in status_result.stdout
+                for i in ["This host is registered", "Registered"]
+            ):
+                return True
+            return False
+
+        assert loop_until(check_registered_with_context)
+        assert result.returncode == 0, f"Registration failed: {result.returncode}"
+
+        # Check no SELinux denials occurred
+        denials = _check_selinux_denials(start_time=start_time)
+        if denials:
+            scontext_matches = re.findall(r"scontext=([^\s]+)", denials)
+            for scontext in scontext_matches:
+                if "insights_client_t" in scontext:
+                    pytest.fail(
+                        f"Process transitioned to confined context {scontext} "
+                        f"instead of staying in unconfined_t. This indicates that "
+                        f"insights-client transitioned to insights_client_t when it "
+                        f"should have remained in unconfined_t. Denials:\n{denials}"
+                    )
+        assert not denials, f"SELinux denials found:\n{denials}"
+
+    finally:
+        # Teardown: Restore original SELinux mode
+        if original_mode != "Disabled":
```

Comment:
This is probably rather a suggestion for future improvement that does not need to be addressed now, but it would make sense to change the `_set_selinux_mode` to a context manager which would during the `__exit__` restore the original state.